core/ledger/kvledger/upgrade_dbs.go

```diff
@@ -44,7 +44,10 @@ func UpgradeDBs(config *ledger.Config) error {
 	}
 
 	if config.StateDBConfig.StateDatabase == "CouchDB" {
-		if err = statecouchdb.DropApplicationDBs(config.StateDBConfig.CouchDB); err != nil {
+		couchdbConfig := config.StateDBConfig.CouchDB
+		couchdbConfig.MaxRetries = 2
+		couchdbConfig.MaxRetriesOnStartup = 2
```

Comment:
Similar to `peer node upgrade-dbs`, the `peer node start` also expects the CouchDB to be up. However, we don't manipulate/change the user configured `MaxRetries` and `MaxRetriesOnStartup` when we execute `peer node start`. I am not comfortable in manipulating the user-configured values only for the  `peer node upgrade-dbs`. It looks like a hack to me.

If the rationale is the following, I am still not comfortable: in v2.0, we allow CouchDB not to be up while executing `upgrade-dbs` bit in v2.2, we need the CouchDB to be up as the peer itself drops CouchDB databases. As admin could have ignored noticing this change and might run this command without keeping CouchDB alive. Hence, we need to return from the `upgrade-dbs` earlier. 

The reason is that the `upgrade-dbs` command would be executed only once when the move is made from v1.4.x to v2.x. It is clear that this command would not be used frequently. If admin makes a mistake for one-time execution, I don't see a strong reason to exit early by manipulating/changing the user-configured `MaxRetries` and `MaxRetriesOnStartup`.

Reply:
@cendhu thank you. This was what we discussed at our scrum. Let's talk again in today's scrum. I am fine to use the user-configured values instead of changing them.

Reply:
Discussed at scrum and agreed to use the user-configured values instead of changing them in individual commands.